src/client/pythonEnvironments/base/locators/lowLevel/fsWatchingLocator.ts

```diff
@@ -14,7 +15,7 @@ import { Locator } from '../../locator';
  * Subclasses can call `this.emitter.fire()` * to emit events.
  */
 export abstract class FSWatchingLocator extends Locator {
-    private initialized = false;
+    private disposables: IDisposable[] | undefined;
```

Comment:
```suggestion
    private readonly disposables = new Disposables();
```
You would not need the dispose method if you use the class.

Or, you can have this class extend `Disposables` as well, making `this.disposable` as protected in the original `Disposables` class. Then you won't even need to define the member variable `disposables`.